go_sequencer/common/l1tx.go

```diff
@@ -144,3 +144,121 @@ func (tx *L1Tx) SetID() error {
 
 	return nil
 }
+
+// L1UserTxFromBytes decodes a L1Tx from []byte
+func L1UserTxFromBytes(b []byte) (*L1Tx, error) {
```

Comment:
We'll need to coordinate with the smart contract team, To ensure the L1UserTx which we'll get from the `Rollup Smart Contract` the positioning of the elements in the []byte is like we are decoding here.

If not already done.